databases/week3/dbweek3.sql

```diff
@@ -0,0 +1,261 @@
+VALUES('Review for Greek salat', 7 , '2021-10-12', 9 );
+INSERT INTO
+  review(title, review_meal_id, created_date, stars)
+VALUES('Review for vanill cake', 8 , '2021-08-19', 1 );
+
+-- Meal - Queries to write
+-- Get all meals
+SELECT 
+  title
+FROM 
+  meal;
+
+-- Get a meal with any id, fx 1
+SELECT 
+  title
+FROM meal
+WHERE id = 2;
+
+-- Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE
+	meal 
+SET
+	price  = 150 
+WHERE
+	id = 1; 
+
+-- Delete a meal with any id, fx 1
+DELETE FROM 
+  meal
+WHERE 
+  id = 2; 
+
+-- Reservation - Queries to write
+-- Get all reservations
+SELECT 
+  *
+FROM
+  reservation;
+
+-- Get a reservation with any id, fx 1
+SELECT 
+  created_date
+FROM reservation
+WHERE id = 1;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE
+	reservation 
+SET
+	number_of_guests = 4 
+WHERE
+	id = 1; 
+
+-- Delete a reservation with any id, fx 1
+DELETE FROM 
+  reservation
+WHERE 
+  id = 12; 
+
+-- Review - Queries to write
+-- Get all reviews
+SELECT
+  *
+FROM
+  review;
+
+-- Get a review with any id, fx 1
+SELECT 
+  *
+FROM 
+  review
+WHERE id = 1;
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE
+	review 
+SET
+	stars = 5 
+WHERE
+	id = 1; 
+
+-- Delete a review with any id, fx 1
+DELETE FROM 
+  reservation
+WHERE 
+  id = 1; 
+
+-- Additional queries
+-- Functionality
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT
+ *
+FROM
+  meal
+WHERE price < 90;
+
+-- Get meals that still has available reservations
+SELECT
+  meal.title, meal.max_reservations, COUNT(reservation.meal_id) as reservation_amount
+FROM
+  meal
+JOIN 
+  reservation 
+ON meal.id = reservation.meal_id
+GROUP BY meal.title, meal.max_reservations
+HAVING reservation_amount < meal.max_reservations;
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+SELECT 
+  *
+FROM
+  meal
+WHERE title LIKE '%toast';
+
+-- Get meals that has been created between two dates
+SELECT
+  *
+FROM
+ meal
+WHERE
+  created_date BETWEEN '2021-10-11' AND '2021-11-11';
+
+-- Get only specific number of meals fx return only 5 meals
+SELECT
+  *
+FROM
+  meal
+LIMIT 4;
+
+-- Get the meals that have good reviews
+SELECT
+  meal.title
+FROM
+  meal
+JOIN review ON
+  meal.id = review.review_meal_id
+WHERE 
+  review.stars > 6
+GROUP BY meal.title;
+
+-- Get reservations for a specific meal sorted by created_date
+SELECT
+  *
+FROM
+  reservation
+JOIN meal ON 
+  reservation.meal_id = meal.id
+WHERE  meal.title = 'Carbonara'
+ORDER BY reservation.created_date;
+
+-- Sort all meals by average number of stars in the reviews
+SELECT
+ meal.title as meal_title, AVG(review.stars) as average_number
+FROM
+  meal
+JOIN review ON 
+  meal.id = review.review_meal_id
+GROUP BY meal_title
+ORDER BY average_number;
```

Comment:
also here, careful with using the alias to group or filter by.

Reply:
@mateusms01 can you check please? hope I get you right =)